crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -79,7 +85,16 @@ fn diagnostic_for_argument(
             .arg
             .as_ref()
             .map_or_else(|| keyword.range(), Ranged::range),
-    ))
+    ));
+
+    if let Some(ref mut diagnostic) = diagnostic {
+        diagnostic.set_fix(Fix::unsafe_edit(Edit::range_replacement(
+            replacement?.to_string(),
+            diagnostic.range,
+        )));
+    }
```

Comment:
We should expand the rule documentation and explain why the fix is unsafe.

Reply:
The fixes are drop in replacements for keywords. I had them unsafe by default for review. Is there a convention over how to classify if a fix as safe or unsafe?

Reply:
What Micha is referring to is to add a section like https://docs.astral.sh/ruff/rules/unused-import/#fix-safety that describes why the fix is marked as unsafe. You can use existing documentation as a reference, search for "Fix safety" at https://docs.astral.sh/ruff/rules.

I think the reason this is unsafe is because the user would still need to update the references to the argument in the function body.

Reply:
Thanks for the link @dhruvmanila , I feel the fixes are safer in this case since the values to keyword arguments or the attributes are not really referenced anywhere. I have marked the fixes as `safe`.

Reply:
> keyword arguments or the attributes are not really referenced anywhere

I'm not sure I understand this, can you expand? The argument most likely is being used in the function body, I'm referring to those references.

Reply:
The keyword arguments are to construction of a dag. Then the task constructed inside the context manager are automatically attached to the dag. They are not referred by the tasks in the context manager body and are specific to the dag object itself.

Below is a dag called `latest_only` where there are two tasks `latest_only` and `task1` where `task1` depends on `latest_only` and is denoted by overloading `>>` operator. Airflow earlier used to have `schedule_interval` and `timetable` but then they were merged to `schedule` keyword argument which is compatible to accept values and is a drop in replacement.

Before

```
with DAG(
    dag_id="latest_only",
    schedule_interval="daily",
    start_date=datetime.datetime(2021, 1, 1),
    catchup=False,
    tags=["example2", "example3"],
    sla_miss_callback=sla_callback
) as dag:
    latest_only = LatestOnlyOperator(task_id="latest_only")
    task1 = EmptyOperator(task_id="task1")

    latest_only >> task1

```

After fixes

```
with DAG(
    dag_id="latest_only",
    schedule="daily",
    start_date=datetime.datetime(2021, 1, 1),
    catchup=False,
    tags=["example2", "example3"],
    sla_miss_callback=sla_callback
) as dag:
    latest_only = LatestOnlyOperator(task_id="latest_only")
    task1 = EmptyOperator(task_id="task1")

    latest_only >> task1

```

Reply:
Oh right, sorry I misunderstood. These are function call arguments and not function parameters. Yeah, I think these should be safe to fix.